Description: Do not write full file path to avoid reproductibility issue.
 Some packages like ospray uses ispc to generate some optimized headers.
 If ispc write the whole file path it creates reprocutibility issue for these
 packages.
Author: Francois Mazen <mzf@debian.org>
Forwarded: no
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/module.cpp
+++ b/src/module.cpp
@@ -2584,7 +2584,8 @@
         perror("fopen");
         return false;
     }
-    fprintf(f, "//\n// %s\n// (Header automatically generated by the ispc compiler.)\n", fn);
+    auto [directory, filename] = GetDirectoryAndFileName("", fn);
+    fprintf(f, "//\n// %s\n// (Header automatically generated by the ispc compiler.)\n", filename.c_str());
     fprintf(f, "// DO NOT EDIT THIS FILE.\n//\n\n");
 
     // Create a nice guard string from the filename, turning any
@@ -2697,7 +2698,8 @@
     FILE *f = DHI->file;
 
     if (DHI->EmitFrontMatter) {
-        fprintf(f, "//\n// %s\n// (Header automatically generated by the ispc compiler.)\n", DHI->fn);
+	auto [directory, filename] = GetDirectoryAndFileName("", DHI->fn);
+	fprintf(f, "//\n// %s\n// (Header automatically generated by the ispc compiler.)\n", filename.c_str());
         fprintf(f, "// DO NOT EDIT THIS FILE.\n//\n\n");
     }
     // Create a nice guard string from the filename, turning any
